android/app/build.gradle

```diff
@@ -7,6 +7,11 @@ buildscript {
     }
 }
 
+project.ext.envConfigFiles = [
+    debug: ".env.staging",
+    release: ".env.production"
+]
```

Comment:
You have to tell the `react-native-config` module which config file to use for which build. If you don't, it defaults to `.env` which would explain that report we got.